### PR TITLE
[16.0] [MIG] queue_job and test_queue_job

### PR DESCRIPTION
**Pull request by baimont**
https://github.com/OCA/queue/issues/459

### CODE REVIEW

File: queue_job/tests/test_model_job_channel.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def test_channel_complete_name_uniq(self):
         # not uses here due to its 'flush()' method inside it and exception raises
         # before the line 'self.env["base"].flush()'. So, we are expecting an IntegrityError.
```
Comment (hugosantosred) [inline queue_job/tests/test_model_job_channel.py]: Maybe this comment has to reference new definition of flush, with the flush_all

```suggestion
        # before the line 'self.env.flush_all()'. So, we are expecting an IntegrityError.
```

**Review by hugosantosred — COMMENTED** [review]
Thanks for the migration. Apart from that comment looks good to me

**Review by hugosantosred — APPROVED** [review]

**Comment by guewen** [comment]
Hi, thanks for your work!
Could you migrate `test_queue_job` as well?
If it is not migrated at the same time, we have no way to tell if `queue_job` behaves as expected as most of the tests are there.

**Comment by baimont** [comment]
> Hi, thanks for your work! Could you migrate `test_queue_job` as well? If it is not migrated at the same time, we have no way to tell if `queue_job` behaves as expected as most of the tests are there.

Hi @guewen, done

File: requirements.txt [hunk 2]
```diff
@@ -1 +1,3 @@
 # generated from manifests external_dependencies
+mock
```
Comment (guewen) [inline requirements.txt]: @baimont can't we replace the import with `from unittest import mock`?

Comment (baimont) [reply requirements.txt]: @guewen yes this is better. I adapted the commit

**Review by guewen — APPROVED** [review]
Great! Many thanks!

**Comment by guewen** [comment]
@baimont did you check if jobs were executed? I'm testing locally (I have been enthusiastic about merging 😅), but jobs aren't executed. Now, I may have botched my install

**Comment by baimont** [comment]
> @baimont did you check if jobs were executed? I'm testing locally (I have been enthusiastic about merging 😅), but jobs aren't executed. Now, I may have botched my install

Ahahaha I always execute them myself with pytest-odoo before pushing any change, so yes

**Comment by guewen** [comment]
That's nice, but sorry I speak about jobs ;)

**Comment by baimont** [comment]
> That's nice, but sorry I speak about jobs ;)

Sorry I misread.
I verified through the web client with https://github.com/acsone/queue/blob/6caca1aa057bc38706af3f92c33ee51c5c6c2c7f/queue_job/controllers/main.py#L110 but not sure that's enough.

**Comment by okuryan** [comment]
@pedrobaeza could you, please, publish this repo on odoo app store?

**Comment by pedrobaeza** [comment]
That should be @dreispt or @simahawk if I don't remember bad.

**Comment by okuryan** [comment]
@dreispt , @simahawk could you, please, publish queue_job to odoo app store?

**Comment by sbidoul** [comment]
@baimont could you also look for commits in 14.0 that where not in the 15.0 branch? The `oca-port` tool may help for that.

For instance #443 is missing, it seems.

**Comment by baimont** [comment]
> @baimont could you also look for commits in 14.0 that where not in the 15.0 branch? The `oca-port` tool may help for that.
> 
> For instance #443 is missing, it seems.

this pr is merged so i guess this has to be done in a new pr

**Comment by dreispt** [comment]
@okuryan @pedrobaeza This repo is already published in Odoo Apps.
See for example: https://apps.odoo.com/apps/modules/15.0/queue_job/

**Comment by pedrobaeza** [comment]
@dreispt you have to publish each branch of each repo...

**Comment by okuryan** [comment]
@dreispt please, let me know when it is published for 16.0 ...

**Comment by okuryan** [comment]
@pedrobaeza are their any ways to reach person responsible for publishing? Or maybe how I can help to speed up publishing process?

**Comment by pedrobaeza** [comment]
Sorry, I can't say, as I'm not on OCA board, and don't know other OCA apps store managers.

**Comment by dreispt** [comment]
I tried again and we can now publish 16.0 branches,
It was done for the Queue repo, and this module is now available:
https://apps.odoo.com/apps/modules/16.0/queue_job/

**Comment by okuryan** [comment]
@dreispt yes, I had to write to Odoo support for this and push them through our account manager. They had bug on Odoo app store not allowing to publish Odoo 16

**Comment by baimont** [comment]
> @baimont could you also look for commits in 14.0 that where not in the 15.0 branch? The `oca-port` tool may help for that.
> 
> For instance #443 is missing, it seems.

I did a fw port of #443 here : https://github.com/OCA/queue/pull/470
And then I saw this : https://github.com/OCA/queue/pull/453

**Comment by sbidoul** [comment]
@baimont I think I meant #433, actually. Sorry for the confusion.

**Pull request merged**